Fix path issue for when transform_path uses a full url.

rbatta opened the pull request:
this should prevent api paths to have `/http://some-url.com/#{path}` and
  simply have `http://some-url.com/#{path}`
- version and changelog bump

djspencer commented on CHANGELOG.md:
```diff
@@ -1,3 +1,6 @@
+## 0.2.10
+- Fix issue where setting a URL when overriding `transform_path` would get a `/` preceding the `http://`
```
Change "would get a `/` preceding the `http://`"
To: "results in a `/` character preceding the `http://`"

djspencer commented on lib/swagger/docs/generator.rb:
```diff
@@ -91,6 +91,12 @@ def generate_doc(api_version, settings, config)
 
         private
 
+        def check_for_http_path(debased_path, api_version)
```
Lets change this method name to something like `transform_http_path`

djspencer commented on lib/swagger/docs/generator.rb:
```diff
@@ -91,6 +91,12 @@ def generate_doc(api_version, settings, config)
 
+        def check_for_http_path(debased_path, api_version)
+          path = "#{Config.transform_path(trim_leading_slash(debased_path), api_version)}"
+          return path if path.include?("http")
```
I think this should be more specific with a regex.
e.g.
`
return path if path.match(/^https?:\/\//)
`

djspencer reviewed (commented):
See inline comments.
Also you introduced a number of white-space changes.
Lets revert those so the diff only shows actual code changes.

Commit:
- fix path issue for when transform_path uses a full url.
  this should prevent api paths to have `/http://some-url.com/#{path}` and
  simply have `http://some-url.com/#{path}`
- version and changelog bump
- incorporate changes from review
